src/System.Net.Sockets/src/System/Net/Sockets/AcceptOverlappedAsyncResult.Windows.cs

```diff
@@ -33,8 +33,12 @@ internal override object PostCompletion(int numBytes)
                 IntPtr remoteAddr;
 
                 // set the socket context
+                bool refAdded = false;
                 try
                 {
+                    _listenSocket.SafeHandle.DangerousAddRef(ref refAdded);
+                    IntPtr handle = _listenSocket.SafeHandle.DangerousGetHandle();
```

Comment:
Nit: I'd feel better about this if _listenSocket.SafeHandle were first stored into a local, and then that local used for all three of the DangerousAddRef/GetHandle/Release calls.  It's not obvious that _listenSocket._handle can't change concurrently between the first and last access.